app/src/components/ProposalList.tsx

```diff
@@ -16,13 +16,18 @@ type ProposalListState = {
 
 class ProposalList extends React.Component<ProposalListProps, ProposalListState> {
   componentDidMount (): void {
-    this.componentDidUpdate();
+    this.shouldComponentUpdate();
   }
 
-  componentDidUpdate (): void {
+  shouldComponentUpdate (): boolean {
     this.props.drizzle.contracts.Governance.methods.getProposals().call()
-      .then((ids: any) => this.setState({ ids: ids }))
+      .then((ids: any) => {
+        this.setState({ ids: ids });
+        return true;
```

Comment:
I am skeptical on the scope of this return, i.e. I think this returns true in the inner function

```
(ids: any) => {
...
}
```

rather than returning from the function scope of `shouldComponentUpdate` the effect of this might end up always return false from `shouldComponentUpdate`

I might be wrong

Reply:
I think James is right, as when I run this branch locally the UI gets stuck on "Loading proposals...".